src/middlewared/middlewared/utils/disk_stats.py

```diff
@@ -0,0 +1,52 @@
+import contextlib
+import logging
+import os
+
+from .disks import get_disk_names, get_disks_with_identifiers
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_disk_stats(disk_identifier_mapping: dict | None = None) -> dict[str, dict]:
+    disk_identifier_mapping = disk_identifier_mapping or get_disks_with_identifiers()
+    available_disks = get_disk_names()
+    stats = {}
+    disk_stats_content = ''
+    with contextlib.suppress(IOError):
+        with open('/proc/diskstats', 'r') as f:
+            disk_stats_content = f.read()
```

Comment:
Let's not read the entire contents of this file into memory. Instead we should iterate over it. (i.e.
```
with open('/proc/diskstats', 'r') as f:
    for line in f:
        parts = line.strip().split()
        if len(parts) < 14 or parts[2] not in available_disks:
            continue